src/scripts/main.js

```diff
@@ -1,3 +1,285 @@
+      && !mergedCells[kRow][kCol]) {
+      board[kRow][kCol] *= 2;
+      mergedCells[kRow][kCol] = true;
+
+      if (board[kRow][kCol] === 2048) {
+        messageWin.classList.remove('hidden');
+        playGame = false;
+      }
+      score += board[kRow][kCol];
+      board[row][col] = 0;
+
+      const cellMerged = fieldCells[kRow * 4 + kCol];
+
+      cellMerged.classList.add('field_cell--merge');
+
+      cellMerged.addEventListener('animationend', () => {
+        cellMerged.classList.remove('field_cell--merge');
+        updateBoard();
+      });
+    } else {
+      kRow -= rowChange;
+      kCol -= colChange;
+      board[kRow][kCol] = board[row][col];
+
+      if (kRow !== row || kCol !== col) {
+        board[row][col] = 0;
+
+        const cellMoved = fieldCells[row * 4 + col];
+
+        if (colChange === 1) {
+          cellMoved.classList.add('field_cell--moveRight');
+
+          setTimeout(() => {
+            cellMoved.classList.remove('field_cell--moveRight');
+            updateBoard();
+          }, 200);
+        }
+
+        if (colChange === -1) {
+          cellMoved.classList.add('field_cell--moveLeft');
+
+          setTimeout(() => {
+            cellMoved.classList.remove('field_cell--moveLeft');
+            updateBoard();
+          }, 200);
+        }
+
+        if (rowChange === 1) {
+          cellMoved.classList.add('field_cell--moveDown');
+
+          setTimeout(() => {
+            cellMoved.classList.remove('field_cell--moveDown');
+            updateBoard();
+          }, 200);
+        }
+
+        if (rowChange === -1) {
+          cellMoved.classList.add('field_cell--moveUp');
+
+          setTimeout(() => {
+            cellMoved.classList.remove('field_cell--moveUp');
+            updateBoard();
+          }, 200);
+        }
+      }
+    }
+  }
+}
+
+function moveRight() {
+  for (let row = 0; row < 4; row++) {
+    for (let col = 2; col >= 0; col--) {
+      mergeAndMove(row, col, 0, 1);
+    }
+  }
+}
+
+function moveLeft() {
+  for (let row = 0; row < 4; row++) {
+    for (let col = 1; col < 4; col++) {
+      mergeAndMove(row, col, 0, -1);
+    }
+  }
+}
+
+function moveDown() {
+  for (let col = 0; col < 4; col++) {
+    for (let row = 2; row >= 0; row--) {
+      mergeAndMove(row, col, 1, 0);
+    }
+  }
+}
+
+function moveUp() {
+  for (let col = 0; col < 4; col++) {
+    for (let row = 1; row < 4; row++) {
+      mergeAndMove(row, col, -1, 0);
+    }
+  }
+}
+
+function resetMergedCells() {
+  for (let row = 0; row < 4; row++) {
+    for (let col = 0; col < 4; col++) {
+      mergedCells[row][col] = false;
+    }
+  }
+}
+
+document.addEventListener('keydown', events => {
+  if (playGame) {
+    const pressedKey = events.key;
+
+    if (pressedKey === 'ArrowUp') {
+      moveUp();
+    } else if (pressedKey === 'ArrowDown') {
+      moveDown();
+    } else if (pressedKey === 'ArrowLeft') {
+      moveLeft();
+    } else if (pressedKey === 'ArrowRight') {
+      moveRight();
+    }
```

Comment:
Use [`switch case`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) in cases where you constantly compare one value with others